util/tls/tls_socket.cc

```diff
@@ -579,42 +582,21 @@ void TlsSocket::AsyncReq::HandleOpAsync() {
     case Engine::NEED_READ_AND_MAYBE_WRITE:
       MaybeSendOutputAsyncWithRead();
       break;
-    // TODO handle NEED_WRITE
+    case Engine::NEED_WRITE:
+      MaybeSendOutputAsync();
```

Comment:
if we get `Engine::NEED_WRITE` I would argue we should call here `StartUpstreamWrite`

Reply:
and MaybeSendOutputAsync is redundant.